tools/metadata_tools/process_metadata.py

```diff
@@ -0,0 +1,138 @@
+        return "/p:Configuration=Debug;Platform=iPhone"
+    else:
+        return "/p:Configuration=Debug,Platform=\"Any CPU\"  /p:AppxPackageSigningEnabled=false"
+
+def write_build_script(list_of_samples, platform, output_dir):
+    '''
+    output_dir: platform-specific output folder containing sample solutions
+    list_of_samples: flat list of sample formal names; should correspond to directories
+    '''
+    output_string = "@echo on"
+    output_string += "\nREM Set up environment variables for Visual Studio."
+    output_string += "\ncall \"C:\\Program Files (x86)\\Microsoft Visual Studio\\2017\\Enterprise\\Common7\\Tools\\VsDevCmd.bat\""
+    output_string += "\nREM ==================================="
+    for sample in list_of_samples:
+        output_string += f"\nCALL %ESRI_SAMPLES_TEMP_BUILD_ROOT%\\nuget restore {sample}\\{sample}.sln -PackagesDirectory %ESRI_SAMPLES_TEMP_BUILD_ROOT%\\packages"
+    output_string += "\n@echo on"
+    
+    for sample in list_of_samples:
+        output_string += f"\nREM Building: {sample}"
+        output_string += f"\nmsbuild {plat_to_msbuild_string(platform)} /clp:errorsonly /flp2:errorsonly;logfile=%ESRI_SAMPLES_TEMP_BUILD_ROOT%\\{platform}\\build.log;append {sample}\\{sample}.sln"
+    
+    file_name = os.path.join(output_dir, "BuildAll_CSharp.bat")
+
+    with open(file_name, 'w+') as output_file:
+        output_file.write(output_string)
+
+def write_samples_toc(platform_dir, relative_path_to_samples, samples_in_categories):
+    '''
+    sample_in_categories is a dictionary of categories, each key is a list of sample_metadata
+    platform_dir is where the readme.md file should be written
+    '''
+    readme_text = "# Table of contents\n\n"
+
+    for category in samples_in_categories.keys():
+        readme_text += f"## {category}\n\n"
+        for sample in samples_in_categories[category]:
+            readme_text += f"* [{sample.friendly_name}]({relative_path_to_samples}/{sample.category}/{sample.formal_name}/readme.md) - {sample.description}\n"
+        readme_text += "\n"
+    
+    readme_path = os.path.join(platform_dir, "../..", "readme.md")
+    with open(readme_path, 'w+') as file:
+        file.write(readme_text)
+
+def main():
+    '''
+    Usage: python3 process_metadata.py {operation} {path_to_samples (ends in src)} {path_to_secondary}
+    Operations: toc; secondary path is empty
+                improve; secondary path is common readme
+                sync; keep metadata in sync with readme
+    '''
+
+    if len(sys.argv) < 3:
+        print("Usage: python3 process_metadata.py {operation} {path_to_samples (ends in src)} {path_to_secondary}")
+        print("Operations are toc, improve, and sync; secondary path is path to common readme source for the improve operation.")
+
+    operation = sys.argv[1]        
+    sample_root = sys.argv[2]
+    common_dir_path = ""
+    if operation == "improve":
+        common_dir_path = sys.argv[3]
```

Comment:
Maybe add error handling here. The script crashes when that argument is missing.